### PR TITLE
Convert lettuce-5.0 tests from groovy to java

### PR DESCRIPTION
**Pull request by jaydeluca**
Related to #7195 

Notes:
* I had to modify the assertions in `LettuceSyncClientTest.testConnectException()` slightly because the groovy test references a class directly that is not accessible in java (io.netty.channel.AbstractChannel.AnnotatedConnectException)
* I created a helper method to handle provisioning the containers and connections for the individual tests that require them. I am using the cleanup extension to defer the closing of the connection, and then also using try with resources in the individual tests. If this seems redundant let me know and I can remove one of them.

### CODE REVIEW

**Comment by laurit** [comment]
@jaydeluca see https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/9574

File: instrumentation/lettuce/lettuce-5.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/lettuce/v5_0/LettuceReactiveClientTest.java [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+                    testing.runWithSpan(
+                        "callback",
+                        () -> {
+                          assertThat(res).isEqualTo(defaultVal);
+                          future.complete(res);
+                        });
+                  });
+        });
+
+    await().untilAsserted(() -> assertThat(future).isCompletedWithValue(defaultVal));
```
Comment (laurit) [inline instrumentation/lettuce/lettuce-5.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/lettuce/v5_0/LettuceReactiveClientTest.java:147]: usually we use
```
assertThat(future.get(10, TimeUnit.SECONDS)).isEqualTo(defaultVal));
```

File: instrumentation/lettuce/lettuce-5.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/lettuce/v5_0/LettuceReactiveClientTest.java [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+        () -> {
+          reactiveCommands
+              .get("NON_EXISTENT_KEY")
+              .defaultIfEmpty(defaultVal)
+              .subscribe(
+                  res -> {
+                    testing.runWithSpan(
+                        "callback",
+                        () -> {
+                          assertThat(res).isEqualTo(defaultVal);
```
Comment (laurit) [inline instrumentation/lettuce/lettuce-5.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/lettuce/v5_0/LettuceReactiveClientTest.java:141]: I suspect that assertions might not work well in these callbacks, if the assertion fails the exception probably just gets swallowed. Could you check what happens if this assertion fails, if no useful info is provided you can just remove it.

Comment (jaydeluca) [reply instrumentation/lettuce/lettuce-5.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/lettuce/v5_0/LettuceReactiveClientTest.java:141]: the tests fail when these assertions fail

image

File: instrumentation/lettuce/lettuce-5.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/lettuce/v5_0/LettuceReactiveClientTest.java (hunk 1 above)
Comment (jaydeluca) [reply instrumentation/lettuce/lettuce-5.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/lettuce/v5_0/LettuceReactiveClientTest.java]: updated, thanks!

File: instrumentation/lettuce/lettuce-5.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/lettuce/v5_0/LettuceReactiveClientTest.java (hunk 2 above)
Comment (laurit) [reply instrumentation/lettuce/lettuce-5.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/lettuce/v5_0/LettuceReactiveClientTest.java]: thanks for checking

**Review by laurit — APPROVED** [review]

**Pull request merged**